nixos/tests/fail2ban.nix

```diff
@@ -0,0 +1,24 @@
+import ./make-test-python.nix ({ pkgs, ... }: {
+  name = "fail2ban";
+
+  nodes.machine = _: {
+    services.fail2ban = {
+      enable = true;
+
+      bantime-increment.enable = true;
+    };
+
+    services.openssh.enable = true;
+  };
+
+  testScript = ''
+    import time
+
+    machine.wait_for_unit("multi-user.target")
+
+    # Let fail2ban start and read its config
+    time.sleep(5)
```

Comment:
I don't understand the need of `time.sleep`, `wait_for_unit` will wait for the unit?

Reply:
Well it waits for the unit to start, but if the configuration is incorrect, the unit still starts correctly but will fail a few moments later

Reply:
Ugh, cannot we have the fail2ban check its configuration in ExecPre or something and fail *before* startup?

Reply:
The only way to check if the config is working is to start the server...

Reply:
ffs, now I really want this to be patched but let's put it aside
The problem is that your test is going to be flakey with probability 1 on Hydra :(

Reply:
Soooo, it seems I don't know how to read and there is an option `--test` to check the config

Reply:
Good :)

Reply:
Though it does not seem to catch all errors

Reply:
Can you investigate/open an issue upstream for the delta?

Reply:
(if it's relevant)